src/main/java/ogc/rs/apiserver/router/routerbuilders/EntityRouterBuilder.java

```diff
@@ -74,19 +78,39 @@ public final Router getRouter() {
     JsonObject oasJson = routerBuilder.getOpenAPI().getOpenAPI();
 
     /* Set the OpenAPI spec route */
-    router.get(getOasApiPath()).handler(routingContext -> {
-      HttpServerResponse response = routingContext.response();
-      response.putHeader("Content-type", "application/vnd.oai.openapi+json;version=3.0");
-      response.send(oasJson.toBuffer());
-    });
-    
+    router
+        .get(getOasApiPath())
+        .handler(
+            routingContext -> {
+              HttpServerResponse response = routingContext.response();
+              String queryParam = routingContext.request().getParam("f");
+              if ("html".equals(queryParam)) {
+                String apiDocFilePath = "docs/apidoc.html";
+                try {
+                  String apiDocContent =
+                      new String(
+                          Files.readAllBytes(Paths.get(apiDocFilePath)), StandardCharsets.UTF_8);
```

Comment:
It might be better to use the vertx.filesystem().readfileblocking (same as in stac catalog code - https://github.com/datakaveri/ogc-resource-server/blob/adeb7dbdf98ddd7bd5c050691c0146085aa240c7/src/main/java/ogc/rs/apiserver/ApiServerVerticle.java#L996).  Unless there is some specific reason the java file stuff is used.